### PR TITLE
Sync model

### PR DESCRIPTION
**Pull request by deep-diver**
- `alignment`'s `get_quantization_config` to get quantization config matching to the ones used in fine-tuning
- reused the [practices](https://github.com/huggingface/alignment-handbook/blob/70769f9e9ba41c7f08ba6c4ff3725441b68b7ca3/scripts/run_sft.py#L111) from `alignment` to set `torch_dtype` and `model_kwargs`

### CODE REVIEW

File: src/gen/local_lm.py [hunk 1]
```diff
@@ -15,14 +19,19 @@ def get_model(model_id, model_revision, load_in_8bit, load_in_4bit, model_args,
...
+    torch_dtype = (
+        model_args.torch_dtype if model_args.torch_dtype in ["auto", None] else getattr(torch, model_args.torch_dtype)
+    )    
+    quantization_config = get_quantization_config(model_args)
+    model_kwargs = dict(
+        revision=model_revision,
+        trust_remote_code=model_args.trust_remote_code,
+        use_flash_attention_2=model_args.use_flash_attention_2,
+        torch_dtype=torch_dtype,
+        device_map=get_kbit_device_map() if quantization_config is not None else None,
```
Comment (sayakpaul) [inline src/gen/local_lm.py]: Even when quantization_config is None user may want to set this to auto to run on GPUs that don't have huge VRAMs.

Comment (deep-diver) [reply src/gen/local_lm.py]: then just "auto" would be fine?

Comment (sayakpaul) [reply src/gen/local_lm.py]: Having "auto" default sounds reasonable to me. Or we could let the user pass it.

Comment (deep-diver) [reply src/gen/local_lm.py]: done! "auto" for now

**Comment by deep-diver** [comment]
@sayakpaul more thoughts?

**Comment by sayakpaul** [comment]
No all good.

**Pull request merged**